app/models/invoice_subscription.rb

```diff
@@ -13,6 +13,15 @@ class InvoiceSubscription < ApplicationRecord
   monetize :subscription_amount_cents, disable_validation: true, allow_nil: true
   monetize :total_amount_cents, disable_validation: true, allow_nil: true
 
+  INVOICE_TYPE = {
+    starting: 'starting',
+    periodic: 'periodic',
+    terminating: 'terminating',
+    charge: 'charge',
```

Comment:
starting, periodic, terminating are kind of related to a lifecycle. I find that `charge` stands out and I dunno what to expect in this invoice.

Reply:
I agree the naming is confusing. I'm currently turning it this way:

```ruby
  INVOICING_REASONS = {
    subscription_starting: 'subscription_starting',
    subscription_periodic: 'subscription_periodic',
    subscription_terminating: 'subscription_terminating',
    in_advance_charge: 'in_advance_charge',
  }.freeze
```